index.js

```diff
@@ -168,7 +213,80 @@ app.all('*', (req, res) => {
     }
   }
 
-  const errorMessage = `Request ${req.url} didn't match any registered route.`;
+  const obfuscatedReqBodyForHash = JSON.parse(JSON.stringify(req.body));
+  recordingsContext.deleteBodyAttributesForHash.forEach((path) => {
+    deleteNestedProperty(obfuscatedReqBodyForHash, path);
+  });
+
+  const [_, host, ...routeParts] = req.url.split('/');
+
+  const headers = {
+    ...req.headers,
+    ...(recordingsContext.forwardHeadersForRoute.find((forwardHeaders) => req.url.startsWith(forwardHeaders.route))
+      ?.headers || {}),
+  };
+
+  const dataToHash = {
+    url: req.url,
+    body: obfuscatedReqBodyForHash,
+    method: req.method,
+    headers: headers,
+  };
+
+  const hash = getShaFromData(dataToHash);
+
+  if (recordingsContext.active) {
+    try {
+      const route = `/${routeParts.join('/')}`;
+      const targetUrl = `https://${host}${route}`;
+      console.log('Proxying from ', req.url, ' to', targetUrl, ' body: ', req.body);
+      const proxyRes = await fetch(targetUrl, {
+        method: req.method,
+        headers: { ...headers, 'Access-Control-Allow-Origin': '*' },
+        ...(req.method !== 'GET' && req.method !== 'HEAD' && req.body ? { body: JSON.stringify(req.body) } : {}),
+      });
+      const status = proxyRes.status;
+
+      res.status(proxyRes.status);
+      const contentType = proxyRes.headers.get('content-type');
+      let body;
+      if (contentType && contentType.includes('application/json')) {
+        body = await proxyRes.json();
+        res.setHeader('Content-Type', 'application/json');
+        res.json(body);
+      } else {
+        body = await proxyRes.text();
+        res.send(body);
+      }
+
+      if (!recordings[hash]) {
+        recordings[hash] = [];
+      }
+      recordings[hash].push({
+        body: typeof body === 'string' ? body : JSON.stringify(body),
+        status,
+        hashData: {
```

Comment:
I think that to call it "request" would be more informative. Would make data in recordings folder more legible.